wwwapp/auth.py

```diff
@@ -26,29 +27,12 @@ def login_view(request):
         del request.session['merge_access_info']
 
     if request.user.is_authenticated:
-        try:
-            access = request.user.accountaccess_set.all()[0]
-        except IndexError:
-            access = None
-        else:
-            client = access.api_client
-            user_info = client.get_profile_info(raw_token=access.access_token)
-
-            user = request.user
-            try:
-                user_profile = UserProfile.objects.get(user=user)
-            except UserProfile.DoesNotExist:
-                new_user_info = UserInfo()
-                new_user_info.save()
-                user_profile, just_created = UserProfile.objects.get_or_create(user=user, user_info=new_user_info)
-
-                # I'm not sure if this condition is necessary.
-                if just_created:
-                    standarize_user_info(user_info)
-                    if 'gender' in user_info:
-                        user_profile.gender = user_info['gender']
-                    user.save()
-                    user_profile.save()
+        # This should never happen if the login flow worked correctly but I'm leaving this here just in case there are any broken users in the database already
+        # (aka I'm too afraid to remove it)
+        user_profile, just_created = UserProfile.objects.get_or_create(user=request.user)
+        if just_created:
+            # TODO: untested - this should trigger the Django admin email thing?
+            logging.getLogger('django').error('User profile was missing for {}. This should have never happened.')
```

Comment:
```suggestion
            mail_admins('User profile was missing for {}. This should have never happened.')
```

from django.core.mail.mail_admins

Reply:
I don't think that sends the trace with request data, logged in user info etc., does it?

Reply:
And since you mentioned you are not actually getting these emails, here is what I mean:
```
Invalid HTTP_HOST header: 'wwasdw.warsztsfdatywww.pl'. You
may need to add 'wwasdw.warsztsfdatywww.pl' to ALLOWED_HOSTS.

Report at /
Invalid HTTP_HOST header: 'wwasdw.warsztsfdatywww.pl'. You may
need to add 'wwasdw.warsztsfdatywww.pl' to ALLOWED_HOSTS.

Request Method: GET
Request URL: https://wwasdw.warsztsfdatywww.pl/
Django Version: 1.11.22
Python Executable: /home/app/.pyenv/versions/3.7.4/bin/uwsgi
Python Version: 3.7.4
Python Path: ['.', '',
'/home/app/.pyenv/versions/3.7.4/lib/python37.zip',
'/home/app/.pyenv/versions/3.7.4/lib/python3.7',
'/home/app/.pyenv/versions/3.7.4/lib/python3.7/lib-dynload',
'/home/app/.pyenv/versions/3.7.4/lib/python3.7/site-packages']
Server time: Wt, 3 Mar 2020 19:33:14 +0000
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'allaccess',
 'crispy_forms',
 'django_select2',
 'django_bleach',
 'compressor',
 'wwwapp',
 'django_cleanup')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')

Request information:
USER: [unable to retrieve the current user]

GET: No GET data

POST: No POST data

FILES: No FILES data

COOKIES: No cookie data

META:
HTTP_ACCEPT = '*/*'
HTTP_CONNECTION = 'close'
HTTP_HOST = 'wwasdw.warsztsfdatywww.pl'
HTTP_USER_AGENT = 'curl/7.68.0'
HTTP_X_FORWARDED_FOR = '[removed]'
HTTP_X_FORWARDED_PROTO = 'https'
HTTP_X_ORIGINAL_URI = '/'
HTTP_X_REAL_IP = '[removed]'
PATH_INFO = '/'
QUERY_STRING = ''
REMOTE_ADDR = '127.0.0.1'
REMOTE_PORT = '12494'
REQUEST_METHOD = 'GET'
REQUEST_URI = '/'
SCRIPT_NAME = ''
SERVER_NAME = 'warsztatywww.pl'
SERVER_PORT = '8000'
SERVER_PROTOCOL = 'HTTP/1.1'
UWSGI_ROUTER = 'http'
uwsgi.core = 1
uwsgi.node = b'warsztatywww.pl'
uwsgi.version = b'2.0.18'
wsgi.errors = <_io.TextIOWrapper name=2 mode='w'
encoding='UTF-8'>
wsgi.file_wrapper = ''
wsgi.input = <uwsgi._Input object at 0x7fb88f02bc50>
wsgi.multiprocess = False
wsgi.multithread = True
wsgi.run_once = False
wsgi.url_scheme = 'https'
wsgi.version = '(1, 0)'

Settings:
Using settings module wwwapp.settings
[removed a lot of settings here to make sure there are no secret keys leaking or something]
```

Reply:
Don't you need to use `.exception` for providing a stacktrace?

Reply:
No idea - all I could find is that Django with default logging configuration will deliver these emails for anything with level >= ERROR under the "django" logging key if DEBUG is disabled, I don't have email sending configured locally to test it

Reply:
I'm pretty sure it won't.

Reply:
https://stackoverflow.com/questions/29392281/manually-trigger-django-email-error-report

Reply:
We need to either:
1) Use above
2) Craft custom message
3) Raise an exception

Reply:
I don't really see the point of doing this manually.. Django has proper logging framework for a reason. Maybe it would be better to configure a custom logger, but it just seems like too much of a hassle for one message.

I fixed the solution we have now and made sure it works.